File: test/poll/dotnet/OnNewEmailPoll.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

using Azure.Connectors.Sdk.Office365.Models;
using Microsoft.Azure.Functions.Worker;
using Microsoft.Azure.Functions.Worker.Extensions.Connector;
using Microsoft.Extensions.Logging;

namespace ConnectorPollSample;

public sealed class OnNewEmailPoll(ILogger<OnNewEmailPoll> logger)
{
    [Function("OnNewEmailPoll")]
    public void Run(
        [ConnectorTrigger(
            DeliveryMode = ConnectorTriggerDeliveryMode.Poll,
            Connection = "ConnectorNamespace",
            PollingEndpoint = "%OnNewEmailEndpoint%",
            IsBatched = true,
            MaxBatchSize = 4,
            Concurrency = 4)]
        ConnectorEvent<Office365OnNewEmailTriggerPayload>[] emails)
    {
        foreach (ConnectorEvent<Office365OnNewEmailTriggerPayload> email in emails)
        {
            logger.LogInformation(
                "Poll email trigger message {MessageId} received: {Payload}",
                email.MessageId,
                System.Text.Json.JsonSerializer.Serialize(email.Data));
        }
    }
}

File: test/poll/nodejs/src/functions/onNewEmailPoll.ts
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

import { app, InvocationContext } from '@azure/functions';

app.connectorTrigger('OnNewEmailPoll', {
    deliveryMode: 'Poll',
    connection: 'ConnectorNamespace',
    pollingEndpoint: '%OnNewEmailEndpoint%',
    cardinality: 'many',
    maxBatchSize: 4,
    concurrency: 4,
    handler: async (inputs: unknown[], context: InvocationContext) => {
        for (const input of inputs) {
            const payload = typeof input === 'string' ? JSON.parse(input) : input;
            context.log(`Poll connector payload: ${JSON.stringify(payload)}`);
        }
    },
});
